### PR TITLE
Use runtime initial data in ScalarAdvection and Burgers

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

The other systems will follow in later PRs. I have all the systems without an EOS done. They basically all follow the same general format.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Systems/Burgers/BoundaryConditions/DirichletAnalytic.cpp [hunk 1]
```diff
@@ -20,7 +38,72 @@ DirichletAnalytic::get_clone() const {
...
+          *u = get<Burgers::Tags::U>(analytic_solution_or_data->variables(
+              coords, tmpl::list<Burgers::Tags::U>{}));
+          (void)time;
+        }
+      });
+  flux_impl(flux_u, *u);
+  return {};
+}
+
+void DirichletAnalytic::fd_ghost(
```
Comment (nilsdeppe) [inline src/Evolution/Systems/Burgers/BoundaryConditions/DirichletAnalytic.cpp:67]: Note: These functions are just moved from the hpp and switched to using the member variable analytic prescription

File: src/Evolution/Executables/ScalarAdvection/EvolveScalarAdvectionFwd.hpp [hunk 2]
```diff
@@ -5,13 +5,11 @@
 
 #include <cstddef>
 
-namespace ScalarAdvection {
-namespace Solutions {
+namespace ScalarAdvection::Solutions {
```
Comment (kidder) [inline src/Evolution/Executables/ScalarAdvection/EvolveScalarAdvectionFwd.hpp]: do you still need these forward declarations?

File: src/Evolution/Executables/Burgers/EvolveBurgersFwd.hpp [hunk 3]
```diff
@@ -15,5 +15,4 @@ class Step;
 }  // namespace Solutions
 }  // namespace Burgers
 
-template <typename InitialData>
```
Comment (kidder) [inline src/Evolution/Executables/Burgers/EvolveBurgersFwd.hpp:18]: again, do you need the forward declarations for the solutions?

**Comment by nilsdeppe** [comment]
Rebased on develop and removed forward decls. Thanks for catching that!

**Review by kidder — APPROVED** [review]

**Pull request merged**